### PR TITLE
Origin paths support for RPM spec file analysis 

### PR DESCRIPTION
**Pull request by zlav**
# Overview

Collect origin path information for RPM spec file analysis

## Acceptance criteria

- Spec files will upload origin path information that will show users in the UI which file an RPM came from.

## Testing plan

- Compile the CLI locally with this branch.
- Analyze a directory that has 2 spec files that have different RPMs in them.
- See in the UI that the origin paths are present.

## Risks

I changed RPM analysis, so I suppose I could have broken that for some people, but I'm not entirely sure how.

## References

[ANE-483](https://fossa.atlassian.net/browse/ANE-483)

## Checklist

- [x] I added tests for this PR's change (or explained in the PR description why tests don't make sense).
- [x] If this PR introduced a user-visible change, I added documentation into `docs/`.
- [x] If this change is externally visible, I updated `Changelog.md`. If this PR did not mark a release, I added my changes into an `# Unreleased` section at the top.
- [x] If I made changes to `.fossa.yml` or `fossa-deps.{json.yml}`, I updated `docs/references/files/*.schema.json`. You may also need to update these if you have added/removed new dependency type (e.g. `pip`) or analysis target type (e.g. `poetry`).

[ANE-483]: https://fossa.atlassian.net/browse/ANE-483?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/Strategy/RPM.hs [hunk 1]
```diff
@@ -77,11 +76,11 @@ findProjects = walkWithFilters' $ \dir _ files -> do
 
   case specs of
     [] -> pure ([], WalkContinue)
-    _ -> pure ([RpmProject dir specs], WalkContinue)
+    _ -> pure (RpmProject dir <$> specs, WalkContinue)
 
 data RpmProject = RpmProject
   { rpmDir :: Path Abs Dir
-  , rpmFiles :: [Path Abs File]
+  , rpmFiles :: Path Abs File
```
Comment (spatten) [inline src/Strategy/RPM.hs:83]: Nit: should we rename this to `rpmFile`, since it's just one file now?

**Review by spatten — APPROVED** [review]
:+1:

File: src/Strategy/RPM.hs (hunk 1 above)
Comment (zlav) [reply src/Strategy/RPM.hs:83]: Yes, good catch

**Comment by saramaebee** [comment]
This was the one I'd looked into most closely. Just to make sure I'm following what you did here, instead of having an RPM project have multiple specfiles, you instead are having each specfile count as its own RPM project, and that's what's giving us granularity over which manifest file accounts for each specific dependency, right?

**Pull request merged**